core/src/main/java/hudson/model/FileParameterValue.java

```diff
@@ -315,7 +316,11 @@ public void write(File to) throws Exception {
 
         @Override
         public void delete() {
-            file.delete();
+            try {
+                Files.deleteIfExists(file.toPath());
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
```

Comment:
Maybe just log instead? I do not think callers would be expecting an unchecked exception here.

Reply:
Just a few lines above, we convert `IOException` into `Error`, so I feel like all bets are off with this code path. I made both that path and this one throw `UncheckedIOException`, as at least that seems better than `Error`.